### ISSUE TITLE
Intermittent `test_save_cfrl` failure

### ISSUE THREAD

**Issue by mauicv** [SeldonIO/alibi#689]
The `test_save_cfrl` test in `alibi/tests/saving.py` has started failing intermittently. This seems to have started occurring as a result of a change in a dependency as the error could not be recreated locally until the pip environment was updated to be in line with [CI] (https://github.com/SeldonIO/alibi/runs/6640017227?check_suite_focus=true) dependencies.

Currently, this is "fixed" by #688 which just raises the tolerance of the `asset_allclose` method.

**Comment by mauicv** [comment]
__Note__: Error only arises after reloading the CFRL_Explainer which makes me think that we change the accuracy of the weight in saving or loading.

**Comment by mauicv** [comment]
difference arises first [here](https://github.com/SeldonIO/alibi/blob/7e8b45d41cf8b7bc0dbf52db7ac9c861ebb901fd/alibi/explainers/backends/tensorflow/cfrl_base.py#L318)

**Comment by ascillitoe** [comment]
@mauicv I imagine you've already tried lots of this, but some things I can think to try based on a similar experience with `alibi-detect` are:

1. Save an explainer. Then re-open it subsequent times. Compare explanations on each load. Are these deterministic? This would rule out if there is some randomness in the loading procedure, versus differences between pre-saved and loaded.
2. If explanations are the same after each load, save multiple explainers. When each of these are re-loaded, are there differences between their explanations? If so, this would suggest there is randomness in saving perhaps.
3.  "change the accuracy of the weight in saving or loading" - this makes me think it might be a `dtype` issue? I would have thought everything is already `float32`, but maybe something defaults to `float64` during re-loading or something?

**Comment by mauicv** [comment]
These are good ideas. I actually did a test in which instead of running the input into the model via the `__call__` method I just ran it through the original and reloaded models layer by layer and check if there was a difference for a particular layer. Both of these results were the same so the difference seems to arise because of the `__call__` method. Internally of `__call__` it does some stuff and then calls `call` which Keras expects you to overide with your layer ops. I'm assuming it's something it does in `__call__`. 

I'd spent quite a while on it by this point and it seems like it's not really a big issue, I'll return to it at some point though.

**Comment by ascillitoe** [comment]
Yeh agreed, the differences are very very small so not a high priority issue!
